src/Multipart/UploadState.php

```diff
@@ -76,6 +94,62 @@ public function setPartSize($partSize)
         $this->partSize = $partSize;
     }
 
+    /**
+     * Set displayProgress. Sends object size to setProgressThresholds.
+     *
+     * @param $totalSize numeric Size of object to upload.
+     */
+    public function setDisplayProgress($totalSize)
+    {
+        $this->setProgressThresholds($totalSize);
+        $this->displayProgress = true;
+    }
+
+    /**
+     * Sets the 1/8th thresholds array. $totalSize is only sent if
+     * 'track_upload' is true.
+     *
+     * @param $totalSize numeric Size of object to upload.
+     *
+     * @return array
+     */
+    public function setProgressThresholds($totalSize)
+    {
+        if(!is_numeric($totalSize)) {
+            throw new \InvalidArgumentException(
+                'The total size of the upload must be a number.'
+            );
+        }
+
+        $this->progressThresholds[0] = 0;
+        for ($i=1;$i<=8;$i++) {
+            $this->progressThresholds []= round($totalSize*($i/8));
+        }
+        return $this->progressThresholds;
+    }
+
+    /**
+     * Prints progress of upload.
+     *
+     * @param $totalUploaded numeric Size of upload so far.
+     */
+    public function getDisplayProgress($totalUploaded)
+    {
+        if (!is_numeric($totalUploaded)) {
+            throw new \InvalidArgumentException(
+                'The size of the bytes being uploaded must be a number.'
+            );
+        }
+
+        if ($this->displayProgress) {
+            while (!empty($this->progressBar)
+                && $totalUploaded >= $this->progressThresholds[0]) {
+                echo array_shift($this->progressBar);
```

Comment:
We should also in the future make this configurable like logfn on TraceMiddleware so that customers can choose where the output goes